### PR TITLE
Add warnings and errors for multuple redirection files

### PR DESCRIPTION
**Pull request by AngryBerryMS**
[AB#341331](https://dev.azure.com/ceapex/Engineering/_workitems/edit/341331)

|Error code|Level|Description|
|---|---|---|
|redirection-file-not-found|Error|A registered file is not found in the repo.|
|source-path-conflict|Error| source_path and source_path_from_root exist at the same time.|
|redirection-path-syntax-error|Warning|Check whether starting with "/" or not.|
|redirection-conflict|Error|Same as before except including sub-redirection files check.|

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6837)

### CODE REVIEW

File: src/docfx/Errors.cs [hunk 1]
```diff
@@ -338,6 +338,27 @@ public static Error CircularRedirection(SourceInfo? source, IEnumerable<FilePath
             /// Behavior: ✔️ Message: ✔️
             public static Error RedirectedFileNotRemoved(FilePath path)
                 => new Error(ErrorLevel.Warning, "redirected-file-not-removed", $"Redirected file '{path}' is still in the repo. After adding a file to the redirection JSON file, you must delete the original file from the repo.");
+
+            /// <summary>
+            /// A redirection file registered in .openpublishing.publish.json is not found in the repo.
+            /// </summary>
+            /// Behavior: ✔️ Message: ✔️
+            public static Error FileNotFound(string path)
+                => new Error(ErrorLevel.Error, "file-not-found", $"Redirection file '{path}' registered in .openpublishing.publish.json is not found in the repo.");
```
Comment (live1206) [inline src/docfx/Errors.cs:347]: we already have `file- not -found`, can we have a different code for this one? Like `redirection-file-not-found`?

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -338,6 +338,27 @@ public static Error CircularRedirection(SourceInfo? source, IEnumerable<FilePath
...
+
+            /// <summary>
+            /// A redirection item cannot contain ‘source_path’ and ‘source_path_from_root’ at the same time.
+            /// </summary>
+            /// /// Behavior: ✔️ Message: ✔️
+            public static Error SourcePathConflict(SourceInfo<string> source)
+                => new Error(ErrorLevel.Error, "source-path-conflict", $"A redirection item cannot contain 'source_path' and 'source_path_from_root' at the same time.", source);
```
Comment (live1206) [inline src/docfx/Errors.cs:354]: It would be better to specify which one is in use in the message.

Comment (AngryBerryMS) [reply src/docfx/Errors.cs:354]: No one will be used when they exist at the same time.

All error codes and messages are confirmed with PM. I think we shouldn't change them.

File: src/docfx/Errors.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/Errors.cs:347]: All error codes and messages are confirmed with PM. I think we shouldn't change them.

**Comment by xichan-ms** [comment]
Some diffs in repos **DevSandbox, learn-pr, test** are expected, for more information, please refer to [PR](https://github.com/dotnet/docfx/pull/6829).

**Review by live1206 — APPROVED** [review]

**Review by yufeih — APPROVED** [review]

**Pull request merged**